sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeDirectoryClient.cs

```diff
@@ -218,7 +218,7 @@ public DataLakeDirectoryClient(Uri directoryUri, AzureSasCredential credential,
         /// The token credential used to sign requests.
         /// </param>
         public DataLakeDirectoryClient(Uri directoryUri, TokenCredential credential)
-            : this(directoryUri, credential.AsPolicy(), null, null)
+            : this(directoryUri, credential.AsPolicy(new DataLakeClientOptions()), null, null)
```

Comment:
we shouldn't allocate new options like here. AsPolicy should be able to take default/null (like we do in blobs).

Reply:
Doesn't it eventually create a new options?

Reply:
it does, good point.